src/client.rs

```diff
@@ -612,16 +612,16 @@ impl RequestBuilder {
     ///
     /// let client = reqwest::Client::new()?;
     /// let res = client.post("http://httpbin.org")
-    ///     .json(&map)
+    ///     .json(&map)?
     ///     .send()?;
     /// # Ok(())
     /// # }
     /// ```
-    pub fn json<T: Serialize>(mut self, json: &T) -> RequestBuilder {
-        let body = serde_json::to_vec(json).expect("serde to_vec cannot fail");
+    pub fn json<T: Serialize>(mut self, json: &T) -> ::Result<RequestBuilder> {
+        let body = serde_json::to_vec(json).map_err(::error::from)?;
         self.headers.set(ContentType::json());
         self.body = Some(Ok(body.into()));
```

Comment:
Also requires an update in the `form` method, I've got a branch working on adjusting the builder to be `&mut self`.